tracer/src/Datadog.Trace/ClrProfiler/InstrumentationDefinitions.cs

```diff
@@ -82,10 +82,7 @@ internal static TraceMethodPayload GetTraceMethodDefinitions()
 
         internal static void Dispose()
         {
-            foreach (var def in Instrumentations)
-            {
-                def.Dispose();
-            }
+            Instrumentations = Array.Empty<NativeCallTargetDefinition2>();
```

Comment:
Nice and easy XD

Reply:
I don't really mind, but note that using Array.Empty here is not beneficial. Because there will only ever be one of those empty arrays created, we still have the same number of allocations in total but we add one more generic type.

Reply:
changed to `null` in https://github.com/DataDog/dd-trace-dotnet/pull/4326/commits/74dbe71587ecf6074f3559b5ccad87ee22ec4cfb.

This is called only once at app shutdown.